src/day8/re_frame/trace.cljs

```diff
@@ -113,80 +115,116 @@
-  )
+  (monkey-patch-reagent))
+
 
-(defn search-input [{:keys [title on-save on-stop]}]
+(defn search-input [{:keys [title on-save on-change on-stop]}]
   (let [val  (r/atom title)
-        save #(let [v (-> @val str clojure.string/trim)]
-                (on-save v))]
+        save #(let [v (-> @val str str/trim)]
+                (when (pos? (count v))
+                  (on-save v)))]
     (fn []
-      [:input {:type        "text"
+      [:input {:style       {:margin-left 7}
+               :type        "text"
                :value       @val
                :auto-focus  true
-               :on-blur     save
-               :on-change   #(reset! val (-> % .-target .-value))
+               :on-change   #(do (reset! val (-> % .-target .-value))
+                                 (on-change %))
                :on-key-down #(case (.-which %)
-                               13 (save)
+                               13 (do
+                                    (save)
+                                    (reset! val ""))
                                nil)}])))
 
+(defn render-traces [padding slower-than-bold-int showing-traces]
+  (doall
+    (for [{:keys [op-type id operation tags duration] :as trace} showing-traces]
+      (let [row-style (merge padding {:border-top (case op-type :event "1px solid lightgrey" nil)})
+            #_#__ (js/console.log (devtools/header-api-call tags))]
 
-(defn render-traces []
-  (let [search           (r/atom "")
+        (list [:tr {:key   id
+                    :style {:color (case op-type
+                                     :sub/create "green"
+                                     :sub/run "#fd701e"
+                                     :event "blue"
+                                     :render "purple"
+                                     :re-frame.router/fsm-trigger "#fd701e"
+                                     nil)}}
+               [:td {:style row-style} (str op-type)]
+               [:td {:style row-style} operation]
+               [:td
+                {:style (merge row-style {:font-weight (if (< slower-than-bold-int duration)
+                                                         "bold"
+                                                         "")})}
+                (.toFixed duration 1) " ms"]]
+              (when true
+                [:tr {:key (str id "-details")}
+                 [:td {:col-span 3} (with-out-str (pprint/pprint (dissoc tags :query-v :event :duration)))]]))))))
+
+(defonce app-state [:traces {:filter-input ""
+                             :filter-items []}]) ;; [{:id (random-uuid) :query "showing" :filter-type "contains"} {:id (random-uuid) :query "Reagent" :filter-type "contains"}
+
+(defn render-trace-panel []
+  (let [filter-input     (r/atom "")
+        filter-items     (r/atom [])
         slower-than-ms   (r/atom "")
-        slower-than-bold (r/atom "")]
+        slower-than-bold (r/atom "")
+        filter-type      (r/atom "contains")]
```

Comment:
Clojure idioms usually have enums as keywords not strings, so `:contains`, not `"contains"`.